docs/frontend-system/architecture/04-plugins.md

```diff
@@ -10,84 +10,151 @@ description: Frontend plugins
 
 ## Introduction
 
-<!--
+In addition to the existing [`plugins`](../../plugins/index.md) documentation let's take a look at what changes for plugins in the new frontend system. If you already created a plugin yourself you will recognise a lot of similarity in the new architecture with the existing one.
 
-Introduce frontend plugins and how each of them is shipped as a separate package. Highlight that each package should only export a single plugin instance.
+Backstage is a single-page application composed of a set of plugins. Each of this plugins should solve exactly one responsibility & follow our suggested [plugin package structure & naming](../../architecture-decisions/adr011-plugin-package-structure.md).
 
--->
+For frontend plugins each plugin should only export a single plugin instance. The new frontend system can detect plugins, if they are exported as `default`, from the plugin package. This is also possible during runtime allowing you to add plugins to your Backstage instance without requiring a restart.
 
 ## Creating a Plugin
 
-<!--
+A plugin is can easily be created, it only requires a plugin `id`:
 
-How to create a simple plugin
-
- -->
-
-```ts
-export const myPlugin = createPlugin({
-  id: 'my-plugin',
+```ts title="plugins/tech-radar/src/index.ts"
+export const techRadarPlugin = createPlugin({
+  id: 'tech-radar',
 });
 ```
 
-<!--
+The created plugin does not yet do anything, it will need some plugin options to be useful:
 
-Note that this plugin is useless in itself, and you need to provide extra options to make it useful, in particular extensions.
-
- -->
+- Extend it's functionality, through pages, navigation items or entity cards, using `extensions`
+- You can make it reachable through `routes`
+- Link to other plugins in Backstage using `externalRoutes`
+- Pass `featureFlags` to it
 
 ### Plugin ID
 
-<!--
+You will decide on the `id` when you [create a plugin](../../). The plugin `id` should follow the respective [naming pattern](./08-naming-patterns.md#plugins) of the new frontend system. To give you an example the GraphiQL plugin has the `id` `tech-radar` & as it follows the naming convention we know that the `techRadarPlugin` is the symbol for the default package export containing the plugin.
 
-link to relevant docs
+### Plugin Extensions
 
--->
+So let's make our plugin a bit more useful! Imagine we want to have a TechRadar plugin that displays the recommended technologies at our organisation, [just like in the Backstage demo instance](https://demo.backstage.io/tech-radar). We want the plugin to be displayed on a page & have a navigation item directing to it. This can be achieved by creating a page extension & a navigation item extension. When adding those to the plugin they will be imported with the plugin package & can be discovered.
```

Comment:
I think this and many of the paragraphs below read perhaps a little bit too much like a guide rather than a description of the architecture. This section of the docs is more geared towards explaining how things work and are wired together, rather than a step-by-step guide to building a plugin.